source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRMediaRenderer.mm

```diff
@@ -43,26 +43,37 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
     std::shared_ptr<Media> mediaElem = std::dynamic_pointer_cast<Media>(elem);
 
     NSMutableDictionary *imageViewMap = [rootView getImageMap];
-    NSString *key = [NSString stringWithCString:mediaElem->GetPoster().c_str() encoding:[NSString defaultCStringEncoding]];
-    UIImage *img = imageViewMap[key];
+    NSString *urlString = [NSString stringWithCString:mediaElem->GetPoster().c_str() encoding:[NSString defaultCStringEncoding]];
+    NSString *numberString = [[NSNumber numberWithUnsignedLongLong:(unsigned long long)(elem.get())] stringValue];
+    NSString *piikey = [NSString stringWithCString:[acoConfig getHostConfig] -> GetMedia().playButton.c_str() encoding:[NSString defaultCStringEncoding]];
+    NSString *piikeyViewIF = [NSString stringWithFormat:@"%llu_playIcon", (unsigned long long)elem.get()];
+
+    NSDictionary *pieces = @{
+        @"number" : numberString,
+        @"url" : urlString,
+        @"playicon-url-imageView" : piikey,
+        @"playicon-url-viewIF" : piikeyViewIF
+    };
+
+    NSString *mediaKey = makeKeyForImage(acoConfig, @"media-poster", pieces);
+    UIImage *img = imageViewMap[mediaKey];
     UIImageView *view = nil;
-    CGFloat heightToWidthRatio = 0.0f;
     ACRContentHoldingUIView *contentholdingview = nil;
 
     // if poster is available, restrict the image size to the width of superview, and adjust the height accordingly
     if (img) {
-        view = [[UIImageView alloc] initWithImage:img];
-
-        if (img.size.width > 0) {
-            heightToWidthRatio = img.size.height / img.size.width;
+        contentholdingview = (ACRContentHoldingUIView *)[rootView getImageView:mediaKey];
+        if (contentholdingview) {
+            view = contentholdingview.subviews[0];
```

Comment:
>ews[0]; [](start = 43, length = 7)

Is there ever a chance that the subviews array is empty so that accessing the first element will result in an error/av? Should we be checking for it upfront or adding an assert condition before trying to index into it?

Reply:
The ACRContentHoldingUIView is a wrapper view to hold a subview such as UIImageView. The content view is created when there is an image view to add to the image view map. before the image view is added to the map, the view is added to the content view.